plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/jdbc/struct/JDBCTableColumn.java

```diff
@@ -200,6 +200,15 @@ public List<DBDLabelValuePair> getValueEnumeration(
         query.append("\nFROM ").append(DBUtils.getObjectFullName(getTable(), DBPEvaluationContext.DML));
         if (valuePattern instanceof String) {
             query.append("\nWHERE ");
+            DBSDataType dataType = DBUtils.getDataType(this);
+            if (dataType != null) {
+                String typeExt = CommonUtils.toString(dataType.geTypeExtension());
+                if ("E".equals(typeExt)) { // for future - maybe not only Enum? But I have no such deep knowledge
+                    // it could be better for PostgreSQL to specify just "varchar" or "text"
+                    // but this not conforms SQL standard, so I left varchar(1000) - it works  
+                    identifier = "cast(" + identifier + " as varchar(1000))";
+                }    
```

Comment:
I don't like the idea of leaking database-specific hacks into the generic model. 

Maybe we can somehow delegate it to `SQLDialect`? Please take a look at `SQLDialect#getCastedAttributeName`.

Reply:
Do we have this bug only in PostgreSQL?

Reply:
I was unable to reproduce it even against PostgreSQL, so I can't claim.

Reply:
Because this is PostgreSQL thing in the first hand - ability to have different data types in different schemas.

Reply:
Not this reason! These are different bugs.
Reproduces clearly, i did it today

Reply:
Yes, seems the issue exists only for PotgreSQL, so it's better to do it dialect-specific.

Reply:
done via dialect specific, please take a look